src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/HttpContextValue.cs

```diff
@@ -0,0 +1,73 @@
+using System.Globalization;
+using System.Text.Encodings.Web;
+using Fluid;
+using Fluid.Values;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Liquid;
+
+namespace OrchardCore.DisplayManagement.Liquid.Values;
+
+internal sealed partial class HttpContextValue : FluidValue
+{
+    private readonly HttpContext _context;
+
+    public override FluidValues Type => FluidValues.Object;
+
+    /// <summary>
+    /// Creates a new instance of a <see cref="HttpContextValue"/> for the specified HTTP context.
+    /// </summary>
+    public HttpContextValue(HttpContext context = null)
+    {
+        ArgumentNullException.ThrowIfNull(context);
+
+        _context = context;
+    }
+
+    public override bool Equals(FluidValue other)
+    {
+        if (other is null)
+        {
+            return false;
+        }
+
+        return ToObjectValue() == other.ToObjectValue();
+    }
+
+    public override bool ToBooleanValue() => true;
+
+    public override decimal ToNumberValue() => 0;
+
+    public override object ToObjectValue() => _context;
+
+    public override string ToStringValue() => _context?.ToString();
+
+#pragma warning disable CS0672 // Member overrides obsolete member
+    public override void WriteTo(TextWriter writer, TextEncoder encoder, CultureInfo cultureInfo)
+#pragma warning restore CS0672 // Member overrides obsolete member
+        => writer.Write(_context?.ToString());
+
+    public async override ValueTask WriteToAsync(TextWriter writer, TextEncoder encoder, CultureInfo cultureInfo)
+        => await writer.WriteAsync(_context?.ToString());
+
+    public override ValueTask<FluidValue> GetValueAsync(string name, TemplateContext context)
+    {
+        var httpContext = _context ?? GetHttpContext(context).ToObjectValue() as HttpContext;
```

Comment:
You can't cache the _context, it's specific to each request. Get a new value of the context every time this method is called. And don't wrap it in an ObjectValue

Reply:
If you want to cache the HttpContext in case the fluid value is used multiple times (not common I'd say, and for what gain). You could use the `TemplateContext` object which has a dictionary. And use a private constant to put it there. But again might be more work that benefits so I wouldn't advise that.

Reply:
> You can't cache the _context, it's specific to each request. Get a new value of the context every time this method is called. And don't wrap it in an ObjectValue

The question again, how ToXXXValue() will be evaluated or should we ignore it for this particular object?

Reply:
I fixed it. Now I understand your question. And now you have your answer.

Reply:
Technically we have to decide what to render when we do `{{ Request }}`. Here it's returning `"Request"`, but we could decide to return the url (even if it's more costly) or something that is valuable. I am fine if you think `""` is actually better in that case, or `"(Request)"` if that helps users understand that it's an object.

Reply:
I'm fine with your changes, I was struggling to display something useful if we render `{{ Request }}` but what you did is much simpler and doesn't require construction injection